### PR TITLE
[server] audit log service

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Introduces generic audit logging, that can be turned on through a feature flag.

It only works for user that have an organizationId. For testing you need to manually update the db entry and then log out and in again.
Just ping me after you signed up and I will do that for you.

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
        <li><b>🏷️ Name</b> - se-audit-logs</li>
        <li><b>🔗 URL</b> - <a href="https://se-audit-logs.preview.gitpod-dev.com/workspaces" target="_blank">se-audit-logs.preview.gitpod-dev.com/workspaces</a>.</li>
        <li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
        <li><b>📦 Version</b> - </li>
        <li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-audit-logs%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,6 +31,8 @@ components/versions/versions.json
 .helm-chart-release
 *.tgz
 
+components/public-api/java/bin
```
Comment (filiptronicek) [inline .gitignore:34]: Feel free to remove and rebase, it's been added in https://github.com/gitpod-io/gitpod/pull/19887 to the respective folder's gitignore.

**Review by filiptronicek — COMMENTED** [review]
Took a quick look at the code and left some comments. I am happy to test and have signed up in the preview environment.

**Comment by Siddhant-K-code** [comment]
> It only works for user that have an organizationId. For testing you need to manually update the db entry and then log out and in again.
> Just ping me after you signed up and I will do that for you.

@svenefftinge can you do that for me 🙏🏼 

temp orgId: `4ac145ee-ad57-431e-9503-2fa5bb745310`
temp userId: `b64e91b6-0a78-4147-ada6-5cb282022ea3`

**Comment by svenefftinge** [comment]
@Siddhant-K-code @filiptronicek   I changed your users to enable audit logs

```
curl -X POST \
  "https://api.$GITPOD_DOMAIN/gitpod.v1.AuditLogService/ListAuditLogs" \
  --header "Authorization: Bearer $TOKEN" \
  --header 'content-type: application/json' \
  --data-raw '{
    "organizationId": "<organization_id>",
    "pagination": {
        "pageSize": 200,
        "page": 0
    },
    "from": "2024-01-13T07:06:45.659Z",
    "to": "2024-07-13T07:06:48.659Z"
}'
```

**Review by filiptronicek — APPROVED** [review]
I can get audit logs correctly for basically everything I did in the preview environment, it's very nice.

I assume the next step is filtering this down to non-personal write events? Do we have a ticket for that?

```json
{
  "pagination": {
    "total": 118
  },
  "auditLogs": [
    {
      "id": "5c62c015-834d-45a6-9090-07716335e22a",
      "timestamp": "2024-06-24T14:16:40.612Z",
      "action": "gitpod.v1.OrganizationService/listOrganizationMembers",
      "organizationId": "1ea2a67e-cc3a-468a-b776-30fe0504f201",
      "actorId": "010529b6-17e5-4289-b9e5-7446e9393164",
      "args": "[{\"pagination\":{\"page\":0,\"token\":\"\",\"pageSize\":1000},\"organizationId\":\"1ea2a67e-cc3a-468a-b776-30fe0504f201\"}]"
    },
    {
      "id": "f1464525-8f33-4f05-97d8-5862dda4bae1",
      "timestamp": "2024-06-24T14:16:40.611Z",
      "action": "gitpod.v1.OrganizationService/getOrganizationInvitation",
      "organizationId": "1ea2a67e-cc3a-468a-b776-30fe0504f201",
      "actorId": "010529b6-17e5-4289-b9e5-7446e9393164",
      "args": "[{\"organizationId\":\"1ea2a67e-cc3a-468a-b776-30fe0504f201\"}]"
    },
    {
      "id": "47dc651d-f6f9-4e20-9491-94584afdb007",
      "timestamp": "2024-06-24T14:16:39.465Z",
      "action": "gitpod.v1.WorkspaceService/listWorkspaces",
      "organizationId": "1ea2a67e-cc3a-468a-b776-30fe0504f201",
      "actorId": "010529b6-17e5-4289-b9e5-7446e9393164",
      "args": "[{\"pinned\":true,\"pagination\":{\"page\":0,\"token\":\"\",\"pageSize\":50},\"searchTerm\":\"\",\"organizationId\":\"1ea2a67e-cc3a-468a-b776-30fe0504f201\"}]"
    },
  ]
}
```

**Pull request merged**